### PR TITLE
Add Credential model

### PR DESCRIPTION
**Pull request by molurin1**

### CODE REVIEW

File: src/foam/nanos/auth/Credential.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+ * from nanopay Corporation.
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'Credential',
+  documentation: 'Model to represent a credential.',
+
+  implements: [
+    'foam.nanos.auth.CreatedAware',
```
Comment (jlhughes) [inline src/foam/nanos/auth/Credential.js:24]: Think we should add ByAware as well. good to know who updated credentials.

File: src/foam/nanos/auth/Credential.js [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+  ],
+
+  properties: [
+    {
+      class: 'String',
+      name: 'id'
+    },
+    {
+      class: 'Boolean',
+      name: 'enable'
```
Comment (jlhughes) [inline src/foam/nanos/auth/Credential.js]: default to true

File: src/foam/nanos/auth/services.jrl [hunk 3]
```diff
@@ -74,3 +74,17 @@ p({
...
     """
 })
+
+p({
+  "class":"foam.nanos.boot.NSpec",
+  "name":"credentialsDAO",
+  "serve":true,
+  "serviceScript":`
+    return new foam.dao.EasyDAO.Builder(x)
+      .setJournalType(foam.dao.JournalType.SINGLE_JOURNAL)
```
Comment (jlhughes) [inline src/foam/nanos/auth/services.jrl:85]: setFuid(true) for id if not provided.

Comment (jlhughes) [reply src/foam/nanos/auth/services.jrl:85]: On second thought, let's not.

Comment (molurin1) [reply src/foam/nanos/auth/services.jrl:85]: Any reason why not?

File: src/foam/nanos/auth/Credential.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+foam.CLASS({
+  package: 'foam.nanos.auth',
+  name: 'Credential',
+  documentation: 'Model to represent a credential.',
+
+  implements: [
+    'foam.nanos.auth.CreatedAware',
+    'foam.nanos.auth.CreatedByAware',
+    'foam.nanos.auth.LastModifiedAware',
```
Comment (jlhughes) [inline src/foam/nanos/auth/Credential.js:27]:
```suggestion
    'foam.nanos.auth.LastModifiedAware',
    'foam.nanos.auth.LastModifiedByAware',
```

File: src/foam/nanos/auth/services.jrl (hunk 3 above)
Comment (jlhughes) [reply src/foam/nanos/auth/services.jrl:85]: I was looking at your config and I think the explicit human ids will be the norm: rbc_credentials, but ids have to be unique, so you can't have 
id: rbc_credentials for two different spids. 
So perhaps it should use FUID and also have a 'service' or 'name' property by which we lookup.  

id: fuid
service: String, like 'RBC_EFT'
then find(EQ(Credential.SERVICE, "RBC_EFT")) and a DAO decorator can add where(ENABLED, TRUE) and we already get Spid filtering.

**Review by jlhughes — APPROVED** [review]

**Pull request merged**